src/movie.rb

```diff
@@ -1,16 +1,17 @@
+require 'date'
 require_relative './item'
 
 class Movie < Item
   attr_accessor :silent
 
-  def initialize(silent:, published_date:, id: Random.rand(1...1000))
-    super(id, published_date)
+  def initialize(silent, publish_date)
+    super(publish_date, archived)
     @silent = silent
   end
```

Comment:
After implementing the `save_data` method for preserving data for your movies, the `publish_date` will be null for every item. To resolve it, you have to add the `@publish_date = publish_date` instance variable to your initializer.